### PR TITLE
#161595854 Add a travel document

### PR DESCRIPTION
**Pull request by ronaldndirangu**
#### What does this PR do?
This is a PR to create travel documents

#### Description of Task to be completed?
- Create endpoint to create travel documents
- Write tests for the create document endpoint
- Add document validator middleware

#### How should this be manually tested?
- Clone the repo - `git clone https://github.com/andela/travel_tool_back.git`
- Enter into the project directory - `cd travel_tool_back`
- Rollback, run database migrations and seed database - `yarn db:rollmigrate`
- Install dependencies and start the server - `make start`
-  [Start](https://github.com/andela/travel_tool_front/pull/278)  the frontend.
- Navigate to [travela](http://travela-local.andela.com:3000/#) in your browser and login
- Grab your token from the application tab in Developers tab
- Open `Postman`
- Add your token in headers
- Navigate to the `Users` table in your database
-  Grab the `userId` of the logged in user from the database user table.
- Make a  request object as shown below in the screenshots and use the userId you obtained. 
- On `postman` Send a `POST` request to `http://localhost:5000/api/v1/documents`

#### Any background context you want to provide?
- The request body should consist of the name field: 
`{
	name : "",
	cloudinary_public_id: "",
	cloudinary_url:"", 
	userId: ""
}`

#### What are the relevant pivotal tracker stories?
[#161595854](https://www.pivotaltracker.com/story/show/161595854)

#### Screenshots (if appropriate)
screen shot 2018-11-09 at 10 08 53

#### Questions:
N/A

### CODE REVIEW

**Comment by Spectrumsun** [comment]
When the image URL and public id are empty the error gets to the database. You should add a validation so that the error does not get to the database.

screenshot 2018-11-12 at 4 54 20 pm 2

File: src/modules/documents/DocumentsController.js [hunk 1]
```diff
@@ -1,8 +1,34 @@
 import models from '../../database/models';
 import CustomError from '../../helpers/Error';
+import Utils from '../../helpers/Utils';
 
 const { Op } = models.Sequelize;
 export default class DocumentsController {
+  static async addDocument(req, res) {
+    try {
+      await models.sequelize.transaction(async () => {
```
[the remark this answers is not in the document]
Comment (Spectrumsun) [reply src/modules/documents/DocumentsController.js]: @ronaldndirangu

**Comment by ronaldndirangu** [comment]
@Spectrumsun Thanks for the feedback. I have made the relevant changes

**Comment by Spectrumsun** [comment]
Good Job @ronaldndirangu

**Pull request merged**